apps/cmp.c

```diff
@@ -2779,6 +2786,58 @@ static void print_status(void)
     OPENSSL_free(buf);
 }
 
+static int do_genm(OSSL_CMP_CTX *ctx)
+{
+    if (opt_infotype == NID_id_it_caCerts) {
+        STACK_OF(X509) *cacerts = NULL;
+
+        if (opt_cacertsout == NULL) {
+            CMP_err("Missing -cacertsout option for -infotype caCerts");
+            return 0;
+        }
+
+        if (!OSSL_CMP_get_caCerts(ctx, &cacerts))
+            return 0;
+
+        /* TODO possibly check authorization of sender/origin */
+        if (cacerts == NULL) {
+            CMP_warn("no CA certificate available");
+        } else if (save_free_certs(cacerts, opt_cacertsout, "CA") < 0) {
+            CMP_err1("Failed to store caCerts from genp in %s", opt_cacertsout);
+            return 0;
+        }
+        return 1;
+    } else {
```

Comment:
Is there really no suitable `opt_infotype` to check for in this branch?  An explicit choice feels safer than a default action.